error during startup: "panic: runtime error: slice bounds out of range"

### Bug report

__System info:__ 

Version is 1.1.0, commit: 800da573

__Steps to reproduce:__

1. We upgraded from influx 0.12.2 which worked ok
2. During startup, we have following error:

```
panic: runtime error: slice bounds out of range

goroutine 627 [running]:
panic(0xa15fe0, 0xc420012110)
        /root/.gvm/gos/go1.7.3/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/influxdb/models.scanTagValue(0xc4f42084e0, 0xd0, 0xd0, 0xd1, 0xffffffffffffff3d, 0xd0, 0xc4f420852a, 0x86)
        /workbench/src/github.com/influxdata/influxdb/models/points.go:1034 +0x90
github.com/influxdata/influxdb/models.parseTags(0xc4f42084e0, 0xd0, 0xd0, 0x0, 0x0, 0x4a)
        /workbench/src/github.com/influxdata/influxdb/models/points.go:1314 +0x243
github.com/influxdata/influxdb/models.ParseKey(0xc4f42084e0, 0xd0, 0xd0, 0xc4f42084e0, 0xd0, 0xd0, 0xd0, 0xd0, 0xd0, 0xc4f4208410)
        /workbench/src/github.com/influxdata/influxdb/models/points.go:195 +0x125
github.com/influxdata/influxdb/tsdb.MeasurementFromSeriesKey(0xc4f4208410, 0xd0, 0xd0, 0xd0)
        /workbench/src/github.com/influxdata/influxdb/tsdb/meta.go:2055 +0x64
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).addToIndexFromKey(0xc435caf400, 0x77de, 0xc4f4208340, 0xd0, 0xd0, 0x2, 0xc420c8ac80, 0x0, 0x0)
        /workbench/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:638 +0xd0
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).LoadMetadataIndex(0xc435caf400, 0x77de, 0xc420c8ac80, 0x0, 0x0)
        /workbench/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:479 +0x57d
github.com/influxdata/influxdb/tsdb.(*Shard).Open.func1(0xc42020cc80, 0x0, 0x0)
        /workbench/src/github.com/influxdata/influxdb/tsdb/shard.go:261 +0x286
github.com/influxdata/influxdb/tsdb.(*Shard).Open(0xc42020cc80, 0xe4bb20, 0xc420044018)
        /workbench/src/github.com/influxdata/influxdb/tsdb/shard.go:275 +0x2f
github.com/influxdata/influxdb/tsdb.(*Store).loadShards.func1(0xc421226240, 0xc4209ce100, 0xc4212262a0, 0xc420c8ac80, 0xc421229114, 0x9, 0xc421258b2e, 0x7,
0xc4212598b6, 0x5)
        /workbench/src/github.com/influxdata/influxdb/tsdb/store.go:196 +0x53b
created by github.com/influxdata/influxdb/tsdb.(*Store).loadShards
        /workbench/src/github.com/influxdata/influxdb/tsdb/store.go:204 +0x499
```
__Expected behavior:__ [What you expected to happen]

Inlfuxdb should load all shards.

__Actual behavior:__ [What actually happened]

Influxdb crashed during startup

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,6 +31,7 @@ The stress tool `influx_stress` will be removed in a subsequent release. We reco
 - [#7615](https://github.com/influxdata/influxdb/issues/7615): Reject invalid subscription urls @allenpetersen
 - [#7396](https://github.com/influxdata/influxdb/issues/7396): CLI should use spaces for alignment, not tabs.
 - [#6527](https://github.com/influxdata/influxdb/issues/6527): 0.12.2 Influx CLI client PRECISION returns "Unknown precision....
+- [#7740](https://github.com/influxdata/influxdb/issues/7740): Fix parse key panic when missing tag value @oiooj
 
 ## v1.1.1 [2016-12-06]
 

--- a/models/points.go
+++ b/models/points.go
@@ -1031,6 +1031,9 @@ func scanTagValue(buf []byte, i int) (int, []byte) {
 		}
 		i++
 	}
+	if i > len(buf) {
+		return i, nil
+	}
 	return i, buf[start:i]
 }
 

--- a/models/points_test.go
+++ b/models/points_test.go
@@ -1976,6 +1976,12 @@ func TestParseKeyEmpty(t *testing.T) {
 	}
 }
 
+func TestParseKeyMissingValue(t *testing.T) {
+	if _, _, err := models.ParseKey([]byte("cpu,foo ")); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+}
+
 func TestPoint_FieldIterator_Simple(t *testing.T) {
 
 	p, err := models.ParsePoints([]byte(`m v=42i,f=42 36`))

